zebra-chain/src/work/tests/prop.rs

```diff
@@ -129,3 +130,205 @@ fn equihash_prop_test_input() -> color_eyre::eyre::Result<()> {
 
     Ok(())
 }
+
+proptest! {
+    /// Check Expanded, Compact, Work, and PartialCumulativeWork conversions.
+    ///
+    /// Make sure the conversions don't panic, and that they round-trip and compare
+    /// correctly.
+    #[test]
+    fn prop_difficulty_conversion(expanded_seed in any::<block::Hash>()) {
+        let expanded_seed = ExpandedDifficulty::from_hash(&expanded_seed);
+
+        let hash_zero = block::Hash([0; 32]);
+        let hash_max = block::Hash([0xff; 32]);
+        prop_assert!(expanded_seed >= hash_zero);
+        prop_assert!(expanded_seed <= hash_max);
+
+        // Skip invalid seeds
+        prop_assume!(expanded_seed != hash_zero);
```

Comment:
If this happens I feel like its a bug in our `block::Hash` Arbitrary impl

Reply:
I'm not sure I understand why this is a bug.

All-zeroes is a valid hash.

Reply:
oh my mistake never mind then